### DISCUSSION TITLE
Adjust Session.LocalChannelDataPacketSize to increase uploading speed

### DISCUSSION THREAD

**Discussion by mryakhont** [sshnet/SSH.NET#1110] [Q&A]
Hi team,

Currently I need to upload a large file (about 1 Gb) to the SFTP server. And I found that the upload speed of SSH.NET sdk is much slower than when using FileZilla. Even if I increase the SftpClient.BufferSize, the uploading speed still not be improved
I investigated and found that the size of the message sent depends on the Session.LocalChannelDataPacketSize variable.
I try to increase this value from 64Kb to 1 Mb and see a significant improvement in speed.

May I ask if this update can cause any potential errors? Is there a limit to this value if I want to increase it to 10 Mb or 50 Mb?

Thanks,

**Comment by darkoperator** [comment]
Max size of a TCP packet is 64k. MTU of the network gear between you and the target can bring that size down and fragment the communication
